src/handlers/oidc.rs

```diff
@@ -115,21 +117,33 @@ broker_handler!(Auth, |app, req| {
     }
 
     if app.providers.contains_key(&email_addr.domain.to_lowercase()) {
-
         // OIDC authentication. Redirect to the identity provider.
         let auth_url = oidc_bridge::request(app, email_addr, client_id, nonce, &parsed_redirect_uri)?;
         Ok(Response::with((status::SeeOther, modifiers::Header(Location(auth_url.to_string())))))
 
     } else {
+        let langtag_de: LanguageTag = "de".parse().unwrap();
+        let accept_language = req.headers.get::<AcceptLanguage>().expect("No accepted language set");
+        let lang_file = if langtag_de.matches(&accept_language[0].item) {
+            "lang/german.mo"
+        } else {
+            "lang/english.mo" // fallback, for easier catalog creation below
+        };
+        let f = File::open(lang_file).expect("could not open the catalog");
+        let catalog = Catalog::parse(f).expect("could not parse the catalog");
 
         // Email loop authentication. Render a message and form.
-        let session_id = email_bridge::request(app, email_addr, client_id, nonce, &parsed_redirect_uri)?;
+        let session_id = email_bridge::request(app, email_addr, client_id, nonce, &parsed_redirect_uri, &catalog)?;
+        
         Ok(Response::with((status::Ok,
                            modifiers::Header(ContentType::html()),
                            app.templates.confirm_email.render(&[
                                ("client_id", &client_id),
                                ("session_id", &session_id),
+                               ("title", &catalog.gettext("Confirm your address")),
+                               ("explanation", catalog.gettext("We've sent you an email to confirm your address.")),
+                               ("use", catalog.gettext("Use the link in that email to login to")),
+                               ("alternate", catalog.gettext("Alternatively, enter the code from the email to continue in this browser tab:")),
```

Comment:
I wish there was a way to distinguish between variables (`client_id` and `session_id`) and translated text in the templates. Is there a gettext helper for our templating system? Can we pass in the whole catalog and thus have something that looks like namespaces?